katsdpimager/report.py

```diff
@@ -0,0 +1,356 @@
+
+        mask = katsdpimager.metadata.target_mask(dataset, target)
+        self.timestamps = dataset.timestamps[mask]
+        self.time_range = bokeh.models.Range1d(
+            datetime.fromtimestamp(self.timestamps[0], timezone.utc),
+            datetime.fromtimestamp(self.timestamps[-1], timezone.utc)
+        )
+        array_ant = dataset.sensor['Antennas/array/antenna'][0]
+        self.elevation = target.azel(timestamp=self.timestamps, antenna=array_ant)[1] << u.rad
+        self.parallactic_angle = target.parallactic_angle(
+            timestamp=self.timestamps, antenna=array_ant) << u.rad
+
+    @property
+    def name(self) -> str:
+        return self.target.name
+
+    @property
+    def description(self) -> str:
+        return self.target.description
+
+    def _add_channel_range(self, fig: bokeh.plotting.Figure) -> None:
+        fig.extra_x_ranges = {'channel': self.channel_range}
+        fig.add_layout(bokeh.models.LinearAxis(x_range_name='channel', axis_label='Channel'),
+                       'above')
+
+    def make_channel_data_source(self) -> bokeh.models.ColumnDataSource:
+        data = {
+            'frequency': self.common.frequencies.to_value(FREQUENCY_PLOT_UNIT),
+            'status': self.status,
+            'noise': self.noise.to_value(FLUX_PLOT_UNIT),
+            'peak': self.peak.to_value(FLUX_PLOT_UNIT)
+        }
+        if self.predicted_natural_noise is not None:
+            predicted_noise = self.predicted_natural_noise * self.normalized_noise
+            data['predicted_noise'] = predicted_noise.to_value(FLUX_PLOT_UNIT)
+        return bokeh.models.ColumnDataSource(data)
+
+    def make_time_data_source(self) -> bokeh.models.ColumnDataSource:
+        data = {
+            'time': [datetime.fromtimestamp(ts, timezone.utc) for ts in self.timestamps],
+            'elevation': self.elevation.to_value(ANGLE_PLOT_UNIT),
+            'parallactic_angle': self.parallactic_angle.to_value(ANGLE_PLOT_UNIT)
+        }
+        return bokeh.models.ColumnDataSource(data)
+
+    def make_plot_status(self, source: bokeh.models.ColumnDataSource) -> bokeh.model.Model:
+        fig = bokeh.plotting.figure(
+            height=200,
+            x_axis_label=f'Frequency ({FREQUENCY_PLOT_UNIT})',
+            y_axis_label='Status',
+            x_range=self.frequency_range,
+            y_range=bokeh.models.FactorRange(factors=['masked', 'failed', 'no-data', 'complete'],
+                                             bounds='auto'),
+            sizing_mode='stretch_width', toolbar_location='below'
+        )
+        fig.cross(x='frequency', y='status', source=source, color=PALETTE[0])
+        self._add_channel_range(fig)
+        return fig
+
+    def make_plot_flux(self, source: bokeh.models.ColumnDataSource) -> bokeh.model.Model:
+        fig = bokeh.plotting.figure(
+            x_axis_label=f'Frequency ({FREQUENCY_PLOT_UNIT})',
+            y_axis_label=f'Flux ({FLUX_PLOT_UNIT})',
```

Comment:
I have a bit of a niggle here about the labeling of the axis as 'Flux'. See my detailed comment below.

Reply:
Relabelled as "Flux density per beam".